### PR TITLE
Fix an issue regarding rejects valid for dcompute when using `-checka…

### PR DESCRIPTION
**Pull request by thewilsonator**
…ctioncontext`

### CODE REVIEW

File: tests/codegen/dcompute_check_action.d [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Check that we ignore the conversion of `assert(x)` -> `assert(x,_d_assert_fail(...))`
+// and dont give any errors for the generated code (which is ignored by codegen).
+// REQUIRES: target_NVPTX
+// RUN: %ldc -c %s -mdcompute-targets=cuda-350 -m64 -output-ll -mdcompute-file-prefix=check_action -output-o -checkaction=context
+// xRUN: FileCheck %s --check-prefix=LL < dcompute_check_action.ll
```
Comment (JohanEngelen) [inline tests/codegen/dcompute_check_action.d:5]: why the FileCheck?  (note that this line is disabled due to __x__RUN)

File: gen/semantic-dcompute.cpp [hunk 2]
```diff
@@ -81,6 +81,10 @@ struct DComputeSemanticAnalyser : public StoppableVisitor {
   }
 
   void visit(VarDeclaration *decl) override {
+    if (!decl->canTakeAddressOf()) {
```
[the remark this answers is not in the document]
Comment (thewilsonator) [reply gen/semantic-dcompute.cpp:84]: Yes, unused global scope scope enum strings should be ignored.

[the remark this answers is not in the document]
Comment (JohanEngelen) [reply gen/semantic-dcompute.cpp:84]: Can you add a comment about this? It's still not so obvious to me, also not obvious that the testcase is testing that.

[the remark this answers is not in the document]
Comment (thewilsonator) [reply gen/semantic-dcompute.cpp:84]: It hits it in some formatting code in the bowels of druntime for formatting asserts generated by `-checkaction=context`

**Comment by thewilsonator** [comment]
Better solution: https://github.com/dlang/dmd/pull/15625

**Pull request closed without merging**